packages/in_app_purchase/lib/src/store_kit_wrappers/sk_request_maker.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter/services.dart';
+import 'package:in_app_purchase/src/channel.dart';
+import 'sk_product_wrapper.dart';
+
+/// A request maker that handles all the requests made by SKRequest subclasses.
+///
+/// There are multiple [SKRequest] subclasses handling different requests in the [StoreKit] with multiple delegate methods,
+/// we consolidateded all the [SKRequest] subclasses into this class to make requests handling more straightforward.
+/// [SKRequest](https://developer.apple.com/documentation/storekit/skrequest?language=objc)
+class SKRequestMaker {
+  /// Starts a request to get SKProduct response.
+  ///
+  /// [productIdentifiers] should contain legit product identifiers that you decleared for the products in the Itunes Connect.
+  /// Duplicate values in [productIdentifiers] will be omiited. If [productIdentifiers] is null, an ["storekit_invalid_argument"] error will
+  /// be returned. If [productIdentifiers] is empty, a [SkProductResponseWrapper] will still be returned with [products] being null.
+  ///
+  /// [SkProductResponseWrapper] is returned if there is no error during the request.
+  /// An exception can be thrown if an underlining error occurs when making the request from the IOS platform.
+  Future<SkProductResponseWrapper> startProductRequest(
+      List<String> productIdentifiers) async {
+    final Map<dynamic, dynamic> productResponseMap = await channel.invokeMethod(
+      '-[InAppPurchasePlugin startProductRequest:result:]',
+      productIdentifiers,
+    );
+    if (productResponseMap == null) {
+      throw PlatformException(
```

Comment:
I'm not sure if throwing an error here is the right answer. This would crash the app, right? And it's completely outside of the developers control.